### ISSUE TITLE
[BUG]: "Has new articles" - but actually doesn't

### ISSUE THREAD

**Issue by Owyn** [martinrotter/rssguard#618]
### Brief description of the issue

![image](https://user-images.githubusercontent.com/1309656/152201882-8a334a86-c26c-47c2-ac3c-d30ba184a679.png)
![image](https://user-images.githubusercontent.com/1309656/152203432-cd93b032-80bd-48e7-915e-6188c466b7b2.png) (all articles already read long ago and no new ones are there as you can see)

### How to reproduce the bug?

This feed: `https://hairywizardztranslations.blogspot.com/feeds/posts/default`

### What was the expected result?

says "new articles" only when there are new articles

### What actually happened?

the opposite

### Other information

If I click "mark selected as read" on the fead - it is still shown as unread, 
but if I click "mark unread" on it first and then mark as read - then it becomes "read", but only before next feed update
[rssguard.log](https://github.com/martinrotter/rssguard/files/7988622/rssguard.log)

### Operating system and version

* OS: Win10 x64
* RSS Guard version:
````
RSS Guard
Version: 4.1.3 (built on Windows/x86_64)
Revision: 03d56b30-nowebengine
Build date: 1/19/22 11:57 AM
Qt: 6.2.2 (compiled against 6.2.2)
````

**Comment by martinrotter** [comment]
happens only with this feed?

**Comment by Owyn** [comment]
> happens only with this feed?

I think so

**Comment by akinokonomi** [comment]
Do you have `Ignore changes in article body when new articles are being fetched` (Settings - Feeds & articles - Articles) checked?

**Comment by martinrotter** [comment]
Found the bug. The feed is of junk quality and some entries DO NOT HAVE TITLES!

```xml
  <entry>
    <id>tag:blogger.com,1999:blog-8518796335108270195.post-7247427995154588504</id>
    <published>2021-10-15T19:38:00.003-07:00</published>
    <updated>2022-01-31T18:12:41.353-08:00</updated>
    <title type='text'></title>
```

RSS Guard cannot add entries with no titles!

**Comment by martinrotter** [comment]
Fixed: 9934786c8
